vscode/src/debugger/session.ts

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import * as vscode from "vscode";
+
+import {
+  ExitedEvent,
+  InitializedEvent,
+  Logger,
+  LoggingDebugSession,
+  TerminatedEvent,
+  logger,
+} from "@vscode/debugadapter";
+import { Subject } from "await-notify";
```

Comment:
The usage of this is pretty trivial. I'd rather not add an external dependency for it. Can you just use a promise?

Reply:
I couldn't figure out how when I first tried. I'll take another look.

Reply:
Tried a new solution. Let me know your thoughts.